### PR TITLE
Deprecate `list.sort` in favor of `sort(x: list)`

### PR DESCRIPTION
**Pull request by jabraham17**
Deprecates the unstable `sort` method of `list`, in favor of using the `sort` procedure in the Sort module.

Resolves https://github.com/chapel-lang/chapel/issues/18100

Summary of work done in this PR:
- Deprecated `list.sort` and moved the implemantion of it to the Sort module
- Updated all tests and documentstion
- Added `proc isSorted(x: list)`
- Added `iter sorted(x: list)`

Testing:
- [x] paratest with/without comm
- [x] built and checked docs

[Reviewed by @lydia-duncan]

### CODE REVIEW

File: test/library/standard/List/sort/isSortedList.chpl [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+  writeln("length 2 list: ", isSorted(lst));
+}
+{
+  var lst = new list([1, 2, 3, 4, 5], locking);
+  writeln("sorted list: ", isSorted(lst));
+}
+{
+  var lst = new list([1, 2, 3, 4, 5, 3], locking);
+  writeln("unsorted list: ", isSorted(lst));
+}
```
Comment (lydia-duncan) [inline test/library/standard/List/sort/isSortedList.chpl:23]: A "first element out of place" and "middle element out of place" test would be good to have here as well

**Review by lydia-duncan — APPROVED** [review]
Yeah, I agree that the use of the list internals is unfortunate but I also don't have a better idea at the moment.  Maybe open an issue about it?

File: test/library/standard/List/sort/isSortedList.chpl (hunk 1 above)
Comment (jabraham17) [reply test/library/standard/List/sort/isSortedList.chpl:23]: Done, and I added a "sorted with duplicates" case as well

**Comment by bradcray** [comment]
:tada:

**Pull request merged**